### PR TITLE
feat(PartnerSites): Allow overriding links for locale switcher

### PR DESCRIPTION
**Pull request by renkoh**
### Motivation
Hirer acquisition would like to use the Partner Sites header for our shopfront redesign. Currently the Partner Sites component has the SEEK Candidate URLs embedded within the component, with locale changes taking the user back to www.seek.com.au. 

### Proposed solution 
In this PR, an additional, optional prop has been added to the partner sites component be able to specify a custom analytics tag, href link and link title. 

current:
```<PartnerSites locale={locale} />```

proposed:
```
localeLinks: {
    AU: {
      'data-analytics': 'header:au+talent',
      href: 'https://talent.seek.com.au',
      title: 'SEEK Employer Australia'
    },
    NZ: {
      'data-analytics': 'header:nz+talent',
      href: 'https://talent.seek.co.nz',
      title: 'SEEK Employer New Zealand'
    }
  }

<PartnerSites locale={locale} localeLinks={localeLinks} />
```

### CODE REVIEW

**Comment by georgespyropoulos** [comment]
What if the API was something like `<PartnerSites locale={locale} business={"candidate" | "employer"} />`

This would save you having to pass in localeLinks and would ensure consistency for those links if other teams want to use this functionality

**Comment by renkoh** [comment]
I think separating out Locales would actually make sense - SEEK Learning and SEEK Business don't have NZ counterparts so don't display the Locale component.

**Comment by renkoh** [comment]
@georgespyropoulos the talent shopfront will be hosted on two domains, one at https://talent.seek.com.au and another at https://www.seek.com.au/employer.

**Comment by renkoh** [comment]
@georgespyropoulos the talent shopfront will be hosted (eventually) under two separate subdomains

**Comment by markdalgleish** [comment]
We should probably update the `PartnerSites` demo to document this API. Would probably be worth adding some snapshot tests, too.

File: react/PartnerSites/PartnerSites.js [hunk 1]
```diff
@@ -7,29 +7,71 @@ import PageBlock from '../PageBlock/PageBlock';
...
+    </Hidden>
+  );
 }
 
 PartnerSites.displayName = 'PartnerSites';
 
+const localeLink = PropTypes.shape({
+  'data-analytics': PropTypes.String,
+  href: PropTypes.String,
+  title: PropTypes.String
```
Comment (michaeltaranto) [inline react/PartnerSites/PartnerSites.js]: These `.string`s should be lowercase i believe.

File: react/PartnerSites/Locales/Locales.js [hunk 2]
```diff
@@ -15,46 +15,41 @@ export default function Locales({ locale, linkRenderer }) {
   });
 
   return (
-    <nav aria-labelledby="Locales" role="navigation" className={styles.root}>
-
+    <nav aria-labelledby='Locales' role='navigation' className={styles.root}>
       <ScreenReaderOnly>
-        <h1 id="Locales">Select your country</h1>
+        <h1 id='Locales'>Select your country</h1>
```
Comment (michaeltaranto) [inline react/PartnerSites/Locales/Locales.js]: We normally use double quotes for html/jsx, not sure if there is a prettier config to solve here.

**Review by tgdev — APPROVED** [review]
Nice work.

Looks like there might be some formatting changes due to lack of prettier config but that could be fixed in separate PR.

**Review by michaeltaranto — APPROVED** [review]

**Pull request merged**